### PR TITLE
Use background delete by default; register deletes of CRs correctly

### PR DESCRIPTION
**Pull request by hausdorff**

### CODE REVIEW

File: pkg/await/await.go [hunk 1]
```diff
@@ -336,6 +336,16 @@ func Deletion(
...
 		deleteOpts.PropagationPolicy = &fg
+	} else {
+		// > 1.7.x. Prior to 1.9.x, the default is to orphan children[1]. Our kubespy experiments
+		// with 1.9.11 is that the controller will actually _still_ mark these resources with the
+		// `orphan` finalizer, although it appears to actuually do background delete correctly. We
+		// therefore set it to background manually, just to be safe.
+		//
+		// nolint
+		// [1] https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/#setting-the-cascading-deletion-policy
+		fg := metav1.DeletePropagationBackground
```
Comment (lblackstone) [inline pkg/await/await.go]: `fg` doesn't make sense as a variable name here. Should probably be `bg` instead.

**Comment by hausdorff** [comment]
The logic of handling the various errors bubbling out of client-go has started to become very subtle, and resulted in a lot of duplicate code across create/delete/update. This PR is important to allow our partners to make progress (in particular on Istio) so I recommend we ship it, but I think we also need to come back and refactor the way we're passing around the discovery client and the client pool.

I've opened an issue to track this work: https://github.com/pulumi/pulumi-kubernetes/issues/281

File: pkg/await/await.go [hunk 2]
```diff
@@ -336,6 +336,16 @@ func Deletion(
 		// 1.6.x option. Background delete propagation is broken in k8s v1.6.
 		fg := metav1.DeletePropagationForeground
 		deleteOpts.PropagationPolicy = &fg
+	} else {
+		// > 1.7.x. Prior to 1.9.x, the default is to orphan children[1]. Our kubespy experiments
+		// with 1.9.11 is that the controller will actually _still_ mark these resources with the
```
Comment (lblackstone) [inline pkg/await/await.go]: s/is that/show that

**Comment by hausdorff** [comment]
@lblackstone any more comments?

**Pull request merged**